docs/usage.md

```diff
@@ -65,6 +65,7 @@ to the pytest.
   bugzilla_password = yourPassword
   ```
 * `--collect-logs` - to collect OCS logs for failed test cases.
+* `--monitoring` - to configure persistent volume on monitoring pods.
```

Comment:
as we will rely on this for different tier tests, I don't think that we should have this option as discussed on our short mtg, let's sync with @mbukatov and check option we already have:
https://github.com/red-hat-storage/ocs-ci/blob/c646ab1f9a87f6af02a74d971c4d08a1ddaecf53/ocs_ci/deployment/deployment.py#L121

Probably this option should be enabled by default if we really need this monitoring.

Reply:
@petr-balogh @Akarsha-rai 

This pull request implements setup of persistent storage for monitoring. This means what it makes sense only if the monitoring itself is enabled.

Right now, neither monitoring nor it's persistent configuration is enabled by default. But when the OCS operator is ready and the OCS setup is finalized, both monitoring and it's persistent setup will be enabled by default, without an option to have it disabled.

I haven't enable monitoring by default because part of the code haven't reached downstream yet, and didn't want to break all other tests without letting the team to evaluate the switch. If you believe that the team is ok with the monitoring enabled under in it's current state, I'm ok with that. Eg. I discussed a dashboard feature which likely requires monitoring with @nehaberry recently.

I would suggest to consider these simple changes:
- rename `--monitoring` option to `--persistent-monitoring`, to avoid clash/confusion with sheer monitoring without additional persistent setup
- make sure that `monitoring_enabled` is set to true when `--persistent-monitoring` is used, or raise an error when this is not the case (because of obvious conflict in this case)
- if team is ok with enabling monitoring by default, let's do that, but keep the option so that one could disable it if it's necessary for testing (it could be useful to be able compare different setups during initial testing with various components integrated together)

I would not try to block this further and trust Akarsha-rai with the persistent setup. In the end, all this including setup of monitoring I introduced in https://github.com/red-hat-storage/ocs-ci/pull/499 will be removed when the initial testing of the OCS operator and OCS installation concludes that it works.

We could also try to set both monitoring and it's persistent setup enabled by default immediately, but I would suggest to do this in another pr, as it could lead to harder and prolongued debugging during testing of this PR.

Reply:
@petr-balogh @mbukatov @keesturam, moved 'persistent-monitoring' to default_config.yaml. So if anyone wants to deploy with persistent-monitoring can pass the value as true in yaml.

Reply:
It looks good and I don't think we need to block it. That said I would like to see a validating jenkins job and related analysis of the results compared to a master result without this change.